qa/src/check_import.c

```diff
@@ -349,5 +317,87 @@ main(int argc, char **argv)
 
     pmiDump();
 
+    sts = pmiEnd();
+    check(sts, "pmiEnd");
+}
+
+void
+test_append(void)
+{
+    int		sts;
+    int		ctx3;
+    int		hdl3;
+
+    ctx3 = pmiStart("myarchive", PMI_FLAG_APPEND);
+    check(ctx3, "pmiStart");
+    sts = pmiAddMetric("my.metric.bar", pmID_build(245, 0, 2), PM_TYPE_U64, pmInDom_build(245,1), PM_SEM_INSTANT, pmiUnits(1,-1,0,PM_SPACE_MBYTE,PM_TIME_SEC,0));
```

Comment:
Why is this pmiAddMetric() needed?  my.metric.bar should have bee

Reply:
The model I'm catering a need for is the sysstat/sadc scenario, where we don't (want to) know whether we are appending or creating - whichever case it is, it should Just Work and do the Right Thing, transparently and correctly.  IOW, the application using libpcp_import doesn't want to know about this subtelty, and the correct behavior should just magically happen.

If the archive doesn't exist create it and populate all metric metadata.  If it does exist, check the metric metadata still matches (in-memory vs on-disk - fail if not) and append to the existing archive.